sentry/src/main/java/io/sentry/metrics/SetMetric.java

```diff
@@ -0,0 +1,45 @@
+package io.sentry.metrics;
+
+import io.sentry.MeasurementUnit;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Sets track a set of values on which you can perform aggregations such as count_unique. */
+@ApiStatus.Internal
+public final class SetMetric extends Metric {
+
+  private final @NotNull Set<Integer> values;
+
+  public SetMetric(
+      final @NotNull String key,
+      final @Nullable MeasurementUnit unit,
+      final @Nullable Map<String, String> tags,
+      final @NotNull Long timestamp) {
+    super(key, unit, tags, timestamp);
+    this.values = new HashSet<>();
+  }
+
+  @Override
```

Comment:
I would add a comment as a Javadoc saying the value will be trimmed to int, as that could be confusing for the user